### ISSUE TITLE
Quantized Gradients + Tweedie Objective Crashes Immediately

### ISSUE THREAD

**Issue by AnotherSamWilson** [microsoft/LightGBM#6134]
## Description
I get crash, reproducible on Windows 11 and a mac M1, when trying to use quantized gradients with the tweedie objective.

The exact error I get on Windows is:
`Process finished with exit code -1073741819 (0xC0000005)`

## Reproducible example
```
import lightgbm as lgb
import numpy as np

dat = np.random.uniform(0,1,size=(100,5))
target = np.random.uniform(0,1,size=100)

dtrain = lgb.Dataset(
    data=dat,
    label=target,
)
params = {
    'objective': 'tweedie',
    'use_quantized_grad': True
}
model = lgb.train(
    params=params,
    train_set=dtrain
)
```

## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell
pip install lightgbm
```

## Additional Comments
This problem goes away if you either change `'use_quantized_grad': False` or use a different objective.

**Comment by shiyu1994** [comment]
@AnotherSamWilson Thanks for using LightGBM.

We just pushed a fix for quantized training (#6092) which has not been included in the released version yet. I've just tested the snippet with the latest commit of the master branch. And the code runs smoothly.

```bash
[90]    train's tweedie: 2.66079
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[91]    train's tweedie: 2.6605
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[92]    train's tweedie: 2.66014
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[93]    train's tweedie: 2.65975
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[94]    train's tweedie: 2.65951
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[95]    train's tweedie: 2.65921
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[96]    train's tweedie: 2.65904
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[97]    train's tweedie: 2.65885
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[98]    train's tweedie: 2.65843
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[99]    train's tweedie: 2.65823
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[100]   train's tweedie: 2.65793
```

**Comment by shiyu1994** [comment]
We'll soon release a new version to include the fix. Sorry for the inconvenience. Before that, you may build the latest version from the latest commit by following the instructions here https://github.com/microsoft/LightGBM/tree/master/python-package#install-from-github. Thanks.
